keps/sig-api-machinery/4595-cel-crd-additionalprintercolumns/README.md

```diff
@@ -0,0 +1,1447 @@
+- Test that validation passes when we create multiple additionalPrinterColumns with both jsonPath and expression fields
+- Test that validation fails when we try to create an additionalPrinterColumn with expression field when the feature gate is turned off
+
+[staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor/tableconvertor_test.go](https://github.com/kubernetes/kubernetes/blob/bc302fa4144d21a338683cd83701661f97be4aba/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/tableconvertor/tableconvertor_test.go)
+
+- Verify that CEL compilation errors are caught at the CRD validation phase
+- Verify that CEL compilation at the TableConvertor creation stage succeeds
+- Verify that TableConvertor is getting created for the CRD with both jsonPath and expression columns
+
+<!-- - `k8s.io/apiextensions-apiserver/pkg/apiserver/schema`: `<date>` - `<test coverage>` -->
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+[test/integration/apiserver/crd_additional_printer_columns_test.go](https://github.com/kubernetes/kubernetes/tree/bc302fa4144d21a338683cd83701661f97be4aba/test/integration/apiserver)
+
+- Verify that CRDs are getting created with additionalPrinterColumns with both jsonPath and expression fields
+- Verify that CEL compilation errors are caught at the CRD validation stage
+- Verify that existing behaviour is not altered when creating CRDs with only jsonPath additionalPrinterColumns
+
+<!-- - <test>: <link to test coverage> -->
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+<!-- - <test>: <link to test coverage> -->
+We will test all cases in integration test and unit test. If needed, we can add e2e tests before beta graduation. We are planning to extend the existing [e2e tests for CRDs](https://github.com/kubernetes/kubernetes/blob/3df3b83226530bda69ffcb7b4450026139b2cd11/test/e2e/apimachinery/custom_resource_definition.go).
```

Comment:
SGTM, this should be fully testable with integration and unit, but we might as well extend the existing CRD tests while we're at it.